pkg/prometheus/prometheus.go

```diff
@@ -382,3 +394,89 @@ func createGauge(
 
 	return gauge, nil
 }
+
+func createHistogram(
+	ctx context.Context,
+	runtime runtime.Runtime,
+	metric *config.Metric,
+	meter otelmetric.Meter,
+) (*Histogram, error) {
+	histogram := &Histogram{Metric: *metric}
+
```

Comment:
> I think the current implementation only works with tracers, right?

Indeed, I will add a check for that. Thanks! 

> More complicated topic: The profile gadgets already output something that looks like a histogram

Great question. I think we have couple of options:

- Run profile gadgets for intervals and transform the [report](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadgets/profile/block-io/types/types.go#L21) using otel sdk.
- Implement our own histogram logic similar to [this](https://github.com/cloudflare/ebpf_exporter/blob/master/exporter/histogram.go) but I am not sure if we should, since that would mean we can't use different exporters in future. 

What do you say?

Reply:
It seems more complicated than I anticipated. I think the issue is that profile gadgets are designed only to show the result after they are done, but prometheus continuously scrapes the data. IMO this is something we can work later on:
1. @flyth is working on metrics collection in eBPF directly, probably that'll fix this as well.
2. I'm working on containerized gadgets and it's very likely that I'll touch the prometheus support there too.

Reply:
I agree with that, it makes sense to revisit this once eBPF metrics / containerized gadgets are supported. For now, I put the check for category then!